### PR TITLE
Better daemon update checking

### PR DESCRIPTION
**Pull request by Mingling94**
Also re-added the siad-import npm commands in the package.json

### CODE REVIEW

File: app/js/daemonManager.js [hunk 1]
```diff
@@ -72,17 +72,21 @@ function DaemonManager() {
 		// duration that it takes to load up the blockchain.
 		apiCall("/daemon/updates/check", function(err, update) {
 			if (err) {
+				console.log(err, update)
 				self.Running = false;
 				// Check again later
-				setTimeout(updatePrompt, 1000);
+				setTimeout(updatePrompt, 5000);
```
Comment (DavidVorick) [inline app/js/daemonManager.js:78]: why 5 seconds?

File: app/js/daemonManager.js [hunk 2]
```diff
@@ -72,17 +72,21 @@ function DaemonManager() {
...
 				self.Running = true;
 				UI.notify("New Sia Client Available: Click to update to " + update.Version, "update", function() {
 					Shell.openExternal('https://www.github.com/NebulousLabs/Sia-UI/releases');
 				});
-			} else {
+			} else if (!update.Available){
 				self.Running = true;
 				UI.notify("Sia client up to date!", "success");
+			} else {
+				console.log("Unkown result from update check!");
```
Comment (DavidVorick) [inline app/js/daemonManager.js]: typo

File: package.json [hunk 3]
```diff
@@ -18,6 +18,11 @@
     "debug": "node-inspector & electron --debug=5858 . & xdg-open http://127.0.0.1:8080/debug?ws=127.0.0.1:8080&port=5858",
     "doc": "jsdoc -c .jsdocrc",
     "lint": "jshint . --verbose --exclude-path=.jshintignore",
+    "postinstall": "npm run sia-repo && npm run sia-import",
```
Comment (DavidVorick) [inline package.json]: this returns errors if you got the Sia repo through ssh

File: package.json [hunk 4]
```diff
@@ -18,6 +18,11 @@
...
+    "sia-repo": "go get -u github.com/NebulousLabs/Sia/...",
```
Comment (DavidVorick) [inline package.json:21]: specifically, this returns errors if you got 'siad' through ssh

File: app/js/daemonManager.js [hunk 5]
```diff
@@ -96,19 +100,13 @@ function DaemonManager() {
 		}, function() {
 			UI.notify('Starting siad...', 'start');
 
-			// daemon as a background process logs output to files
-			var out = Fs.openSync(Path.join(__dirname, siaPath, 'daemonOut.log'), 'w');
-			var err = Fs.openSync(Path.join(__dirname, siaPath, 'daemonErr.log'), 'w');
```
Comment (DavidVorick) [inline app/js/daemonManager.js:101]: is there somewhere else that the files go? I feel like these files would be really useful to have, though I forgot about them and haven't asked users for them yet.

File: package.json [hunk 6]
```diff
@@ -18,6 +18,11 @@
...
+    "sia-import": "(test -d app/dependencies/Sia || mkdir -p app/dependencies/Sia) && cp $GOPATH/bin/siad app/dependencies/Sia",
+    "sia-kill": "killall siad; while pgrep siad; do sleep .5; done;",
```
Comment (DavidVorick) [inline package.json:23]: I'm not sure I understand why this rule is needed

File: app/js/daemonManager.js (hunk 1 above)
Comment (Mingling94) [inline app/js/daemonManager.js:78]: 1 second is too frequent to easily read debug output while 5 seconds isn't really that noticeable of a wait imo

File: package.json [hunk 7]
```diff
@@ -18,6 +18,11 @@
     "debug": "node-inspector & electron --debug=5858 . & xdg-open http://127.0.0.1:8080/debug?ws=127.0.0.1:8080&port=5858",
     "doc": "jsdoc -c .jsdocrc",
     "lint": "jshint . --verbose --exclude-path=.jshintignore",
+    "postinstall": "npm run sia-repo && npm run sia-import",
+    "sia-repo": "go get -u github.com/NebulousLabs/Sia/...",
+    "sia-import": "(test -d app/dependencies/Sia || mkdir -p app/dependencies/Sia) && cp $GOPATH/bin/siad app/dependencies/Sia",
```
Comment (DavidVorick) [inline package.json]: I don't know why we need this rule - it hasn't been needed for installation for Luke + I. But maybe that's because we haven't done any clean installs since the rule got removed.

File: package.json (hunk 4 above)
Comment (Mingling94) [inline package.json:21]: I see, I'll remove the postinstall line but the other commands do no harm to stay

File: app/js/daemonManager.js (hunk 5 above)
Comment (Mingling94) [inline app/js/daemonManager.js:101]: I got an error once from these lines. Since siad is no longer a background process, these messages go to their parent process's console, e.g. the general UI's

File: app/js/daemonManager.js (hunk 1 above)
Comment (DavidVorick) [inline app/js/daemonManager.js:78]: didn't realize it's an update check.

5 seconds is probably more than needed too. Even 10 minutes should be fine as far as software updates go.

Comment (Mingling94) [inline app/js/daemonManager.js:78]: 5 seconds is the time to retry updating if it fails. While this hasn't succeeded, Daemon.Running is false and no api calls are being made.

File: package.json (hunk 6 above)
Comment (Mingling94) [inline package.json:23]: To simulate siad not working/going offline for debugging purposes

File: package.json (hunk 7 above)
Comment (Mingling94) [inline package.json]: How do you update the siad in the UI?

File: app/js/daemonManager.js (hunk 1 above)
Comment (bitspill) [inline app/js/daemonManager.js:78]: Unrelated to the pull at hand, is there a way to check Daemon.Running from a plugin? I couldn't find it.

Comment (Mingling94) [inline app/js/daemonManager.js:78]: If you look at pluginManager.js, I implemented a trick where:
if a plugin's apiCall fails
Daemon.ifSiad() is triggered
    which will flip Daemon.Running if that fails too
    or will send back the error and null result if ifSiad() shows siad is running

You could add another ipc channel specifically for this, though

Comment (DavidVorick) [inline app/js/daemonManager.js:78]: I definitely don't think that you should be flipping `Daemon.Running` based on the outcome from a call to `/daemon/updates/check`. The only api call that should be really used to check if the daemon is working is `/daemon/version`

Comment (Mingling94) [inline app/js/daemonManager.js:78]: Yep, `ifSiad()` does precisely that, turning it false if daemon hasn't started. Then it starts it, waiting until an update check returns true to flip Daemon.Running to true.

I _suppose_ since Luke made the update check pointless that updatePrompt() should just be turned into

```
function checkSiad() {
    ifSiad(function() {
        UI.notify('siad started!', 'check');
    }, function() {
        UI.notify('Waiting for siad', 'ongoing');
        // Check again later
        setTimeout(checkSiad, 1000);
    });
}
```

Comment (Mingling94) [inline app/js/daemonManager.js:78]: @DavidVorick Thoughts on this approach? It would be an entirely different PR
